WordPress/gutenberg
# LinkControl: refined the display of the link preview title and url when both are same

## What?
Fixes: #60900

## Why?
When the link text contains the protocol, the preview displays both lines (text and link) and they contain the same information, the only difference is the displayed protocol. So instead of displaying it twice its better to show it only once.

## How?
This PR adds a check that weather the URL and Text is same after removing the protocol from the text.

## Testing Instructions
1. Add a Paragraph that contains the text `https://test.org`.
2. Select the text `https://test.org` and add a link to it `test.org` without the protocol part.
3. Once the link is added, click the link in the Paragraph to open the link control preview.
4. Observe the link preview displays only one line: test.org.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/77401999/a9d01146-bd23-42af-8f8f-9503d09dd395
